### PR TITLE
Rain now cleans blood and grime off marines and puts out burning mobs

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request
Title. 
Regular rain (but not acid) will clean marines who stand in it.
EDIT; for the sake of consistency water will also extinguish mobs who stand it it, and due to feedback acid will also clean marines, albeit with a different flavor text.

## Why It's Good For The Game
Consistency updates to make the world feel more alive = good

## Changelog
:cl:
add: Rain will now clean marines who stand in it, like a shower.
add: Rain will extinguish players who stand in it, also like a shower.
/:cl:

### CODE REVIEW

**Comment by QualityVan** [comment]
Was going to say isn't there just a proc on mobs to handle that, but no, the behavior is specifically part of showers and space cleaner.

**Comment by Al-1ce** [comment]
If the acid has a high enough pH to melt through metal armor, it can definitely dissolve blood. The difference between messages would add some nice fluff.

File: code/datums/weather/weather_types/acid_rain.dm [hunk 1]
```diff
@@ -64,7 +64,23 @@
...
+		H.update_inv_back()
+	if(H.wear_suit?.clean_blood())
+		H.update_inv_wear_suit()
+	if(H.w_uniform?.clean_blood())
+		H.update_inv_w_uniform()
+	if(H.gloves?.clean_blood())
+		H.update_inv_gloves()
+	if(H.shoes?.clean_blood())
+		H.update_inv_shoes()
+	H.clean_blood()
```
Comment (TiviPlus) [inline code/datums/weather/weather_types/acid_rain.dm:86]:
```suggestion
	if(!ishuman(L))
		return
	var/mob/living/carbon/human/H = L
	if(H.back?.clean_blood())
		H.update_inv_back()
	if(H.wear_suit?.clean_blood())
		H.update_inv_wear_suit()
	if(H.w_uniform?.clean_blood())
		H.update_inv_w_uniform()
	if(H.gloves?.clean_blood())
		H.update_inv_gloves()
	if(H.shoes?.clean_blood())
		H.update_inv_shoes()
	H.clean_blood()
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by Strayed-Raven** [comment]
I do not like this because I want to stay bloody.

File: code/datums/weather/weather_types/acid_rain.dm (hunk 1 above)
Comment (Lumipharon) [reply code/datums/weather/weather_types/acid_rain.dm]: Shower has a fairly detailed proc to cover washing a mob. Would it make more sense to make a wash proc at the carbon level, then some extra stuff at human etc, and have shower/rain just call that?

**Comment by DerpiestBruh** [comment]
flamers go home

**Comment by TiviPlus** [comment]
I dont think I like extinguishing

**Comment by Lumipharon** [comment]
> I dont think I like extinguishing

Could just make it reduce firestacks by X amount. So standing in the rain means you won't be on fire for as long, but doesnt just completely cuck it.

**Comment by RipGrayson** [comment]
> I dont think I like extinguishing

It's not an immediate extinguishing currently, it's a low prob chance which means you can get lucky and the fire will go out immediately or it can take several minutes. What Lumi said might be the best way to go, just adjusting firestacks instead of outright extinguishing.

I would like a decision on if this is an acceptable compromise though, so I have a direction to head with this PR.

**Comment by BraveMole** [comment]
> > I dont think I like extinguishing
> 
> It's not an immediate extinguishing currently, it's a low prob chance which means you can get lucky and the fire will go out immediately or it can take several minutes. What Lumi said might be the best way to go, just adjusting firestacks instead of outright extinguishing.
> 
> I would like a decision on if this is an acceptable compromise though, so I have a direction to head with this PR.

What lumi proposed is better imo

**Comment by bowlofcereal** [comment]
could you make it also clean the map up of blood, like wash away the blood on the ground? a lot of blood and gib decals building up cause a lot of server lag if the round goes on long enough and it's rather tedious to have to go in and delete them all using sdql

File: code/datums/weather/weather_types/acid_rain.dm [hunk 2]
```diff
@@ -52,6 +52,10 @@
 	if(prob(max(0,100-resist)))
 		L.adjustFireLoss(7)
 		to_chat(L, span_boldannounce("You feel the acid rain melting you away!"))
+	cleanup(L, TRUE)
+	if(L.on_fire && prob(10))
+		L.ExtinguishMob()
+		to_chat(L, span_warning("As the acidic rain burns your body away, it also extinguishes the fire with a steamy hiss."))
```
Comment (Lumipharon) [inline code/datums/weather/weather_types/acid_rain.dm:58]:
```suggestion
	if(L.fire_stacks > -20)
		L.fire_stacks = max(-20, L.fire_stacks - 3)
```
Something like this. Also side note, why the hell is normal rain a child of acid rain?

File: code/datums/weather/weather_types/acid_rain.dm [hunk 3]
```diff
@@ -64,7 +68,43 @@
...
-
 	probability = 60
 	repeatable = TRUE
+
+/datum/weather/acid_rain/harmless/weather_act(mob/living/L)
+	cleanup(L)
+	if(L.on_fire && prob(20))
+		L.ExtinguishMob()
+		L.fire_stacks = -30
+		to_chat(L, span_warning("With a hiss, the heavy rain extinguishes the fire on your body, leaving nasty burns behind."))
```
Comment (Lumipharon) [inline code/datums/weather/weather_types/acid_rain.dm:79]:
```suggestion
	if(L.fire_stacks > -20)
		L.fire_stacks = max(-20, L.fire_stacks - 3)
```
same thing

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by RipGrayson** [comment]
Rain now decreases only firestacks and doesn't put out mobs directly, thanks to [Lumipharon](https://github.com/Lumipharon) for suggesting the code involved.

**Comment by Lumipharon** [comment]
@RipGrayson Maybe get rid of the prob(25) for now. 1 stack per second isn't that high considering you'll typically get 10+ put on you, and likely more when you have to run through fire tiles.

We can always put it back if it turns out to be turbo fast.

**Review by Lumipharon — APPROVED** [review]
We must clean

**Pull request merged**